### PR TITLE
Adding build for ARM instances

### PR DESCRIPTION
**Pull request by jonahjon**
wip

Starting the PR for adding a 2nd nodegroup with graviton2 instances for ARM testing.

Signed-off-by: jonahjon <jonahjones094@gmail.com>

### CODE REVIEW

**Comment by jonahjon** [comment]
Next step is to add the Falco testing onto ARM now!

File: config/prow/crier.yaml [hunk 1]
```diff
@@ -60,6 +60,9 @@ spec:
       - name: oauth
         secret:
           secretName: oauth-token
+      nodeSelector:
+        Archtype: "x86"
+
```
Comment (leodido) [inline config/prow/crier.yaml:65]:
```suggestion
```

Whiteline

File: config/prow/hook.yaml [hunk 2]
```diff
@@ -126,6 +126,9 @@ spec:
       - name: plugins
         configMap:
           name: plugins
+      nodeSelector:
+        Archtype: "x86"
+
```
Comment (leodido) [inline config/prow/hook.yaml:131]:
```suggestion
```

Whiteline

File: config/prow/prow-controller-manager.yaml [hunk 3]
```diff
@@ -52,6 +52,9 @@ spec:
         - name: job-config
           configMap:
             name: job-config
+      nodeSelector:
+        Archtype: "x86"
+
```
Comment (leodido) [inline config/prow/prow-controller-manager.yaml:57]:
```suggestion
```

Whiteline

File: config/prow/pushgateway.yaml [hunk 4]
```diff
@@ -22,6 +22,9 @@ spec:
           ports:
             - name: http
               containerPort: 9091
+      nodeSelector:
+        Archtype: "x86"
+
```
Comment (leodido) [inline config/prow/pushgateway.yaml:27]:
```suggestion
```

Whiteline

File: config/prow/statusreconciler.yaml [hunk 5]
```diff
@@ -63,6 +63,9 @@ spec:
       - name: s3-credentials
         secret:
           secretName: s3-credentials
+      nodeSelector:
+        Archtype: "x86"
+
```
Comment (leodido) [inline config/prow/statusreconciler.yaml:68]:
```suggestion
```

Whiteline

**Review by leodido — COMMENTED** [review]
Can't wait to see this completed! 🔥 

P.S.: I only removed some spare white lines

**Comment by maxgio92** [comment]
Tainted ARM EKS Node Group to prevent x86 jobs to be scheduled on it, now that new jobs configs with node selection to ARM / x86 are not already deployed.

**Comment by maxgio92** [comment]
The currently-deployed EBS CSI driver has sidecars with only x86 images. We need to replace them with multiarch images (https://github.com/kubernetes-sigs/aws-ebs-csi-driver/tree/v0.9.0/deploy/kubernetes).
More here: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/604

**Comment by maxgio92** [comment]
Unfortunately the EBS CSI driver 0.9.0 version ARM kustomization still missed the daemonset's liveness sidecar multiarch image.
Opened a PR: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/731

**Comment by jonahjon** [comment]
Both the regular and gprc build passed

https://prow.falco.org/view/s3/falco-prow-logs/logs/arm-test-falco/1358467537897000960/
https://prow.falco.org/view/s3/falco-prow-logs/logs/arm-test-falco-grpc/1358539260709310464/

**Comment by jonahjon** [comment]
I think phase 2 is figure out a way to scale this instance to zero and run both these tests like once a week.... will do another PR though.

**Comment by maxgio92** [comment]
I think we can merge this PR (so we can also remove the `nodeTaint`s as we deploy `nodeSelector`s for `ProwJob`s ).
WDYT?

**Comment by jonahjon** [comment]
> I think we can merge this PR (so we can also remove the node `Taint`s as we deploy `nodeSelector`s for `ProwJob`s ).
> WDYT?

I think that sounds good, let's do it

**Review by maxgio92 — APPROVED** [review]

**Pull request merged**